rosidl_parser/rosidl_parser/definition.py

```diff
@@ -14,32 +14,77 @@
 
 from collections import OrderedDict
 import pathlib
-
-
-"""Basic types as defined by the IDL specification."""
-BASIC_TYPES = [
-    'short',
-    'unsigned short',
-    'long',
-    'unsigned long',
-    'long long',
-    'unsigned long long',
+from typing import Iterable
+from typing import Tuple
+
+# Basic types as defined by the IDL specification
+
+# 7.4.1.4.4.2 Basic Types
+SIGNED_NONEXPLICIT_INTEGER_TYPES = (  # rules (26)
+    'short',  # rule (27)
+    'long',  # rule (28)
+    'long long',  # rule (29)
+)
+UNSIGNED_NONEXPLICIT_INTEGER_TYPES = (  # rules (30)
+    'unsigned short',  # rule (31)
+    'unsigned long',  # rule (32)
+    'unsigned long long',  # rule (33)
+)
+NONEXPLICIT_INTEGER_TYPES = (
+    *SIGNED_NONEXPLICIT_INTEGER_TYPES,
+    *UNSIGNED_NONEXPLICIT_INTEGER_TYPES,
+)
+FLOATING_POINT_TYPES = (  # rule (24)
     'float',
     'double',
     'long double',
-    'char',
-    'wchar',
-    'boolean',
-    'octet',
-    'int8',
-    'uint8',
-    'int16',
-    'uint16',
-    'int32',
-    'uint32',
-    'int64',
-    'uint64',
-]
+)
+CHARACTER_TYPES = (
+    'char',  # rule (34)
+    'wchar',  # rule (35)
+)
+BOOLEAN_TYPE = 'boolean'  # rule (36)
+OCTET_TYPE = 'octet'  # rule (37)
+
+# 7.4.13.4.4 Integers restricted to holding 8-bits of information
+# 7.4.13.4.5 Explicitly-named Integer Types
+SIGNED_EXPLICIT_INTEGER_TYPES = (
+    'int8',  # rule (208)
+    'int16',  # rule (210)
+    'int32',  # rule (211)
+    'int64',  # rule (212)
+)
+UNSIGNED_EXPLICIT_INTEGER_TYPES = (
+    'uint8',  # rule (209)
+    'uint16',  # rule (213)
+    'uint32',  # rule (214)
+    'uint64',  # rule (215)
+)
+EXPLICIT_INTEGER_TYPES = (
+    *SIGNED_EXPLICIT_INTEGER_TYPES,
+    *UNSIGNED_EXPLICIT_INTEGER_TYPES,
+)
+SIGNED_INTEGER_TYPES = (   # rules (26) + (208) + (210-212)
+    *SIGNED_NONEXPLICIT_INTEGER_TYPES,
+    *SIGNED_EXPLICIT_INTEGER_TYPES,
+)
+UNSIGNED_INTEGER_TYPES = (   # rules (30) + (209) + (213-215)
+    *UNSIGNED_NONEXPLICIT_INTEGER_TYPES,
+    *UNSIGNED_EXPLICIT_INTEGER_TYPES,
+)
+INTEGER_TYPES = (   # rules (25) + ()
+    *SIGNED_INTEGER_TYPES,
+    *UNSIGNED_INTEGER_TYPES,
+)
+
+"""All basic types as defined by the IDL specification."""
+BASIC_TYPES = (
+    *INTEGER_TYPES,
+    *FLOATING_POINT_TYPES,
+    *CHARACTER_TYPES,
+    BOOLEAN_TYPE,
+    OCTET_TYPE,
+)
```

Comment:
Providing constants for the various groups of types allows message generators to refer to those (e.g. `INTEGER_TYPES`) instead of needing to enumerate them explicitly.